controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java

```diff
@@ -146,7 +146,7 @@ final class OperationContextImpl extends AbstractOperationContext implements Aut
 
     private final ModelControllerImpl modelController;
     private final OperationMessageHandler messageHandler;
-    private final Map<ServiceName, ServiceController<?>> realRemovingControllers = new HashMap<>();
+    private final Set<ServiceController<?>> realRemovingControllers = new HashSet<>();
```

Comment:
This should also use Collections.newSetFromMap(new IdentityHashMap<>()), no?

Reply:
@ropalka What's your opinion about Paul's comment?

Reply:
Correct @pferraro !